server/plugin.go

```diff
@@ -0,0 +1,373 @@
+		return
+	}
+
+	var messageNotification SNSMessageNotification
+	if err := json.Unmarshal([]byte(notification.Message), &messageNotification); err != nil {
+		p.API.LogDebug("AWSSNS HandleNotification Decode Error on message notification", "err=", err.Error())
+		return
+	}
+
+	p.API.LogDebug("AWSSNS HandleNotification", "MESSAGE", notification.Subject)
+	var fields []*model.SlackAttachmentField
+	fields = addFields(fields, "AlarmName", messageNotification.AlarmName, true)
+	fields = addFields(fields, "AlarmDescription", messageNotification.AlarmDescription, true)
+	fields = addFields(fields, "AWS Account", messageNotification.AWSAccountID, true)
+	fields = addFields(fields, "Region", messageNotification.Region, true)
+	fields = addFields(fields, "New State", messageNotification.NewStateValue, true)
+	fields = addFields(fields, "Old State", messageNotification.OldStateValue, true)
+	fields = addFields(fields, "New State Reason", messageNotification.NewStateReason, false)
+	fields = addFields(fields, "MetricName", messageNotification.Trigger.MetricName, true)
+	fields = addFields(fields, "Namespace", messageNotification.Trigger.Namespace, true)
+	fields = addFields(fields, "StatisticType", messageNotification.Trigger.StatisticType, true)
+	fields = addFields(fields, "Statistic", messageNotification.Trigger.Statistic, true)
+	fields = addFields(fields, "Period", strconv.Itoa(messageNotification.Trigger.Period), true)
+	fields = addFields(fields, "EvaluationPeriods", strconv.Itoa(messageNotification.Trigger.EvaluationPeriods), true)
+	fields = addFields(fields, "ComparisonOperator", messageNotification.Trigger.ComparisonOperator, true)
+	fields = addFields(fields, "Threshold", fmt.Sprintf("%f", messageNotification.Trigger.Threshold), true)
+
+	var dimensions []string
+	for _, dimension := range messageNotification.Trigger.Dimensions {
+		dimensions = append(dimensions, fmt.Sprintf("%s: %s", dimension.Name, dimension.Value))
+	}
+	fields = addFields(fields, "Dimensions", strings.Join(dimensions, "\n"), false)
+
+	msgColor := "#008000"
+	if messageNotification.NewStateValue == "ALARM" {
+		msgColor = "#FF0000"
+	} else if messageNotification.NewStateValue == "INSUFFICIENT" {
+		msgColor = "#FFFF00"
+	}
+
+	attachment := &model.SlackAttachment{
+		Title:  notification.Subject,
+		Fields: fields,
+		Color:  msgColor,
+	}
+
+	post := &model.Post{
+		ChannelId: p.ChannelID,
+		UserId:    p.BotUserID,
+		Props: map[string]interface{}{
+			"from_webhook":      "true",
+			"override_username": SNS_USERNAME,
+			"override_icon_url": SNS_ICON_URL,
+		},
+	}
+
+	model.ParseSlackAttachment(post, []*model.SlackAttachment{attachment})
+	if _, appErr := p.API.CreatePost(post); appErr != nil {
+		return
+	}
+	return
+}
+
+func (p *Plugin) handleUnsubscribeConfirmation(body io.Reader) {
+	// var subscribe SubscribeInput
+	// if err := json.NewDecoder(body).Decode(&subscribe); err != nil {
+	// 	return
+	// }
+	return
```

Comment:
Is there something we should be doing here? Maybe notify someone?

Reply:
@jasonblais @cpanato what was/should be the intent here?